_posts/2019-06-05-uncertainty-estimates-tfprobability/uncertainty_tfprobability.Rmd

```diff
@@ -0,0 +1,782 @@
+---
+title: "Adding uncertainty estimates to Keras models with tfprobability"
+description: > 
+   As of today, there is no mainstream road to obtain uncertainty estimates from neural networks. All that can be said this that normally, approaches tend to be Bayesian in spirit, involving some way of putting a prior over model weights. This holds true as well for the method presented in this post: We show how to use tfprobability, the R interface to TensorFlow Probability, to add uncertainty estimates to a Keras model in an elegant and conceptually plausible way.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019uncertaintytfp
+date: 06-05-2019
+categories:
+  - Probability and statistics 
+  - Keras
+  - Concepts
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/uci_both.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+About six months ago, we showed [how to create a custom wrapper](https://blogs.rstudio.com/tensorflow/posts/2018-11-12-uncertainty_estimates_dropout/) to obtain uncertainty estimates from a Keras network. Today we present a less laborious, as well faster-running, way using [tfprobability](https://rstudio.github.io/tfprobability/), the R wrapper to TensorFlow Probability. Like most posts on this blog, this one won't be short, so let's quickly state what you can expect in return of reading time.
+
+## What to expect from this post
+
+Starting from what _not_ to expect: There won't be a recipe that tells you how _exactly_ to set all parameters involved in order to report the "right" uncertainty measures. But then, what _are_ the "right" uncertainty measures? Unless you happen to work with a method that has no (hyper-)parameters to tweak, there will always be uncertainty about how to report uncertainty.
+
+What you _can_ expect though is an introduction to obtaining uncertainty estimates for Keras networks, as well as an empirical report of how tweaking (hyper-)parameters may affect the results. As in the aforementioned post, we perform our tests on a simulated as well as on a real dataset, the [Combined Cycle Power Plant Data Set](http://archive.ics.uci.edu/ml/datasets/Combined+Cycle+Power+Plant). At the end, in place of strict rules you should have acquired some intuition that will transfer to other real-world datasets.
```

Comment:
```suggestion
What you _can_ expect, though, is an introduction to obtaining uncertainty estimates for Keras networks, as well as an empirical report of how tweaking (hyper-)parameters may affect the results. As in the aforementioned post, we perform our tests on both a simulated and a real dataset, the [Combined Cycle Power Plant Data Set](http://archive.ics.uci.edu/ml/datasets/Combined+Cycle+Power+Plant). At the end, in place of strict rules, you should have acquired some intuition that will transfer to other real-world datasets.
```

Reply:
Restructured the second sentence to use "and" instead of "as well as" (since you used it just above). Also added a comma after "in place of strict rules". 
😳 Sorry, I don't know why the GitHub UI doesn't show specific diffs in some cases.